BasicController.java

```diff
@@ -36,7 +36,7 @@ public ApiResult multiply(@RequestParam BigDecimal multiplier, @RequestParam Big
    @GetMapping("/divide")
    public ApiResult divide(@RequestParam BigDecimal dividend, @RequestParam BigDecimal divisor) {
       if (divisor.equals(BigDecimal.ZERO)) {
-         return getResultFromError("no division by null");
+         return ("no division by null");
```

Comment:
Similarly, in the `divide` method, the return statement inside the if condition is returning a `String` directly instead of an `ApiResult`. This will also cause a type mismatch error.